src/HotChocolate/Core/src/Types/Types/Descriptors/Conventions/DefaultNamingConventions.cs

```diff
@@ -341,8 +341,17 @@ public NameString FormatFieldName(string fieldName)
                 nameof(fieldName));
         }
 
-        return fieldName.Length > 1
-            ? fieldName.Substring(0, 1).ToLowerInvariant() + fieldName.Substring(1)
-            : fieldName.ToLowerInvariant();
+        if(fieldName.Length < 2)
+            return fieldName.ToLowerInvariant();
+
+        var lastUpperCaseIndex = 0;
+
+        for (var i = 0; i < fieldName.Length; i++)
+        {
+            if (char.IsUpper(fieldName[i]))
+                lastUpperCaseIndex = i;
+        }
+
+        return fieldName[..lastUpperCaseIndex].ToLowerInvariant() + fieldName[lastUpperCaseIndex..];
```

Comment:
@michaelstaib  did we already drop support for older versions? so can we use ranges now?

Reply:
Sorry my bad, didn't consider that ppl were using still Standard 2.0 and earlier versions of c#

Reply:
@Nivisan-Jeyananthan  i know, thats the sad part about developing a framework :D 

So basically there are always to different kind of new language features, Compile TIme and Runtime Features.
We can use Compile Time features immediately as these are compiled to compatible code. We can only use runtime features after we drop support for the old frameworks, what basically means, after 2 years </3

Reply:
@PascalSenn wow, makes sense. Thanks for telling me! :)
I was wondering why so many new C# features weren't used in your codebase.